test/C/src/concurrent/Tracing.lf

```diff
@@ -14,7 +14,9 @@ reactor Source {
 		self->s++;
 	=}
 }
-reactor TakeTime {
+reactor TakeTime(
+    bank_index:int(0)
```

Comment:
Does having a default value (other than 0) make sense here? 
Maybe it could be used to indicate the starting index of the `bank_index` in user code.
For example, for the following instantiation:
```
    t = new[4] TakeTime(bank_index=4);
```
the bank indexes will be `{4, 5, 6, 7}`.

To a lesser extent, I think the type might be problematic. A reactor such as
```
reactor Foo(
    bank_index:char*("1")
)
```
is a problem since it cannot be used as a bank. Perhaps, the validator can check this and disallow a type for bank_index altogether?

Reply:
My reasoning is that there is no reason to prevent a user from creating a parameter named `bank_index` if the reactor is never used in a bank.  If it is used in a bank, and the type cannot accept an integer value, then the generated code will fail to compile. I think it would be difficult to check all the cases in the validator because validity may depend on the target language.

Reply:
I think at least in Python, we would not get a compile error. Most likely, the user code will be written expecting that `bank_index` is of a certain type. If the reactor is then later used in a bank, the `bank_index` will become a new type (int), which might or might not result in a runtime error (or an unexpected behavior).

I think the safest solution is to disallow an initial value and type for `bank_index` for all targets. The produced validator error can then explain that `bank_index` is reserved for banks.

Reply:
In that case, I think that a better design would be for bank_index to always automatically be a parameter. The user would never need to declare it.  @cmnrd: How does it work in Cpp?  Does bank_index need to be declared as parameter to be used?  What if it is given the wrong type?

Reply:
Yes, in C++ `bank_index` needs to be declared by the user. I think that this has the advantage that the user can freely choose any integer compatible type. If an incompatible type is specified, the target compiler will complain. This is exactly the same behavior as for 'normal' parameters. Since Python is not type checked and things can go wrong for any parameter, I don't think this is much of a problem. We just need to document it well and tell users not to use `bank_index` in other contexts.

I think that disallowing initial values and parameters for the `bank_index` is a bad idea. Treating it differently would just cause confusion for many users. Also I am not sure how this would solve the issue in Python, as even without an initializer you can explicitly pass any object to `bank_index` when doing a normal instantiation.